src/it/projects/pom-create-lock-file-with-integrity-checking/expected-pom.xml

```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+  xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd"
+  xmlns:integrity="urn:se.vandmo.dependencylock.integrity">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>se.vandmo.tests</groupId>
+  <artifactId>pom-dependency-lock</artifactId>
+  <version>0-SNAPSHOT</version>
+
+  <dependencies>
+    <dependency>
+      <groupId>io.netty</groupId>
+      <artifactId>netty-buffer</artifactId>
+      <version>4.1.65.Final</version>
+      <type>jar</type>
+      <scope>compile</scope>
+      <integrity:checksum>sha256:E7679BA4F3B2FBEC1CD356380B056254FC4E9C0093EF7B083FBF75A79A6FE333</integrity:checksum>
```

Comment:
NPM seems to use sha512 in package-lock.json for some reason. I haven't dug any deeper though, maybe it is because of length extension attacks? I am all for using a fast algorithm that is "secure enough" but is SHA256 that algorithm?

Reply:
If NPM uses SHA-512 then I think the sanest answer is to just copy what they do. I'll update to SHA-512 across this pr.